didcomm/core/authcrypt.py

```diff
@@ -0,0 +1,111 @@
+from dataclasses import dataclass
+from typing import List
+
+from authlib.common.encoding import to_bytes, to_unicode, urlsafe_b64encode, json_dumps, urlsafe_b64decode
+from authlib.jose import JsonWebEncryption
+
+from didcomm.common.algorithms import AuthCryptAlg, Algs
+from didcomm.common.resolvers import ResolversConfig
+from didcomm.common.types import DID_OR_DID_URL, DID_URL
+from didcomm.common.utils import find_key_agreement_secret_and_verification_methods, extract_key, \
+    find_key_agreement_sender_verification_method, find_key_agreement_recipient_secrets, parse_base64url_encoded_json
+from didcomm.errors import MalformedMessageError, MalformedMessageCode
+
+
+@dataclass(frozen=True)
+class AuthcryptResult:
+    msg: bytes
+    to_kids: List[DID_OR_DID_URL]
```

Comment:
Need to return to keys, not just kids. It will be needed for anoncrypt of authcrypted messaged in case protected_sender_id=True.